setup/process_env.py

```diff
@@ -169,24 +137,30 @@ def main():
     if "SECRET_KEY" not in env_vars or not env_vars["SECRET_KEY"]:
         updates["SECRET_KEY"] = generate_secret_key()
 
-    if len(sys.argv) > 1 and sys.argv[1].strip():
-        connection_string = sys.argv[1].strip()
+    connection_string = sys.argv[1].strip() if (len(sys.argv) > 1 and sys.argv[1].strip()) else None
+
+    if connection_string:
+        pg_env_vars = construct_env_vars_from_connection_string(connection_string)
+
     else:
         required_pg_keys = ["POSTGRES_HOST", "POSTGRES_DB", "POSTGRES_USER"]
-        pg_missing = any(key not in env_vars or not env_vars[key] for key in required_pg_keys)
+        is_required_pg_env_vars_missing = any(key not in env_vars or not env_vars[key] for key in required_pg_keys)
 
-        if not pg_missing:
-            # Build a connection string from existing env variables.
-            connection_string = construct_pg_conn_str(env_vars)
-        else:
-            # If any required PG key is missing, ensure we got a connection string as a parameter.
+        if is_required_pg_env_vars_missing:
             sys.exit("Required PostgreSQL connection parameters are missing in the .env file and no connection string was provided.")
 
-    pg_updates = validate_pg_connection_string(connection_string)
-    updates.update(pg_updates)
+        pg_env_vars = {
+            "POSTGRES_HOST": env_vars["POSTGRES_HOST"],
+            "POSTGRES_PORT": env_vars.get("POSTGRES_PORT", ""),
+            "POSTGRES_DB": env_vars["POSTGRES_DB"],
+            "POSTGRES_USER": env_vars["POSTGRES_USER"],
+            "POSTGRES_PASSWORD": env_vars.get("POSTGRES_PASSWORD", "")
+        }
```

Comment:
I don't think we should save any of these, unless they specifically came from a connection string. It doesn't make sense to me to save empty or vacuous values.

Reply:
Also supporting underlying assumptions.